### PR TITLE
Issue #155: Support dashboard lists

### LINKED ISSUE #155: Support for Dashboard Lists
Requesting support for Dashboard Lists.
https://docs.datadoghq.com/api/?lang=bash#dashboard-lists

### PR DESCRIPTION
**Pull request by nyanshak**
Resolve #155 by adding support for dashboard lists:
* get all lists
* get a single list
* create a list
* update a list
* delete a list
* add items to a list
* update items in a list
* get items in a list
* delete items from a list

### CODE REVIEW

**Comment by nyanshak** [comment]
Fix #155 

Still working on adding some tests, but wanted some early feedback :)

@zorkian @yfronto

**Comment by yfronto** [comment]
I think I'd like to see a more user-friendly way to handle dashboard types, but I'm not sure what the best way to do that is. I see `custom_timeboard`, `custom_screenboard`, `integration_screenboard`, `integration_timeboard`, and `host_timeboard`, but I don't see how to pragmatically decide what type a particular dashboard is. I suppose you could create public vars for the types, which could at least limit the possible inputs?

File: dashboard_lists.go [hunk 1]
```diff
@@ -12,12 +12,19 @@ import (
...
+const (
+	DashboardListItemCustomTimeboard        = "custom_timeboard"
+	DashboardListItemCustomScreenboard      = "custom_screenboard"
+	DashboardListItemIntegerationTimeboard  = "integration_timeboard"
+	DashboardListItemIntegrationScreenboard = "integration_screenboard"
+	DashboardListItemHostTimeboard          = "host_timeboard"
+)
+
 // DashboardList represents a dashboard list.
 // struct when we load the summaries.
```
Comment (yfronto) [inline dashboard_lists.go]: This comment is a bit confusing with the period and line break.

File: dashboard_lists.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+// DashboardList represents a dashboard list.
+// struct when we load the summaries.
+type DashboardList struct {
+	Id             *int    `json:"id,omitempty"`
+	Name           *string `json:"name,omitempty"`
+	DashboardCount *int    `json:"dashboard_count,omitempty"`
+}
+
+type DashboardListItem struct {
```
Comment (yfronto) [inline dashboard_lists.go]: We should probably have a docstring on this, since it's exported.

File: dashboard_lists.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+const (
+	DashboardListItemCustomTimeboard        = "custom_timeboard"
+	DashboardListItemCustomScreenboard      = "custom_screenboard"
+	DashboardListItemIntegerationTimeboard  = "integration_timeboard"
+	DashboardListItemIntegrationScreenboard = "integration_screenboard"
+	DashboardListItemHostTimeboard          = "host_timeboard"
+)
+
+// DashboardList represents a dashboard list.
+// struct when we load the summaries.
```
Comment (yfronto) [inline dashboard_lists.go]: This comment is confusing with the line break.

**Comment by yfronto** [comment]
lgtm

**Comment by yfronto** [comment]
Hey @nyanshak, do you want to merge this or shall I?

**Pull request merged**